### PR TITLE
[COST-2147] - Remove AWS mat views

### PR DESCRIPTION
**Pull request by maskarb**
- Delete the old AWS mat views
- Update the necessary migrations so that the SQL for the mat views are no longer run.

### CODE REVIEW

**Review by abaiken — APPROVED** [review]

**Comment by abaiken** [comment]
Going to try to test this in ephemeral

**Review by abaiken — COMMENTED** [review]
Testing this in the ephemeral env, and creating a new source it looks like the mat views are persistent: 
```
koku=# set search_path to acct10002;
SET
koku=# \dm
                                       List of relations
  Schema   |                    Name                     |       Type        |      Owner       
-----------+---------------------------------------------+-------------------+------------------
 acct10002 | reporting_aws_compute_summary               | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_compute_summary_by_account    | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_compute_summary_by_region     | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_compute_summary_by_service    | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_cost_summary                  | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_cost_summary_by_account       | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_cost_summary_by_region        | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_cost_summary_by_service       | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_database_summary              | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_network_summary               | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_storage_summary               | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_storage_summary_by_account    | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_storage_summary_by_region     | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_aws_storage_summary_by_service    | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_compute_summary             | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_cost_summary                | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_cost_summary_by_account     | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_cost_summary_by_location    | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_cost_summary_by_service     | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_database_summary            | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_network_summary             | materialized view | tFcBT6DW983Qv7Ek
 acct10002 | reporting_azure_storage_summary             | materialized view | tFcBT6DW983Qv7Ek
--More--
```

File: koku/reporting/migrations/0213_delete_aws_mat_views.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.1.13 on 2021-12-08 18:38
+from django.db import migrations
+
+
```
Comment (Red-HAP) [inline koku/reporting/migrations/0213_delete_aws_mat_views.py:4]:
```suggestion
from koku.database import get_model


SQL_TMPL = """DROP MATERIALIZED VIEW IF EXISTS {} CASCADE ;"""


def drop_matview_sql(model_str):
    return SQL_TMPL.format(get_model(model_str)._meta.db_table)


```

File: koku/reporting/migrations/0213_delete_aws_mat_views.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        migrations.DeleteModel(name="AWSCostSummary"),
+        migrations.DeleteModel(name="AWSCostSummaryByAccount"),
+        migrations.DeleteModel(name="AWSCostSummaryByRegion"),
+        migrations.DeleteModel(name="AWSCostSummaryByService"),
+        migrations.DeleteModel(name="AWSDatabaseSummary"),
+        migrations.DeleteModel(name="AWSNetworkSummary"),
+        migrations.DeleteModel(name="AWSStorageSummary"),
+        migrations.DeleteModel(name="AWSStorageSummaryByAccount"),
+        migrations.DeleteModel(name="AWSStorageSummaryByRegion"),
+        migrations.DeleteModel(name="AWSStorageSummaryByService"),
```
Comment (Red-HAP) [inline koku/reporting/migrations/0213_delete_aws_mat_views.py:23]:
```suggestion
        migrations.RunSQL(sql=drop_matview_sql("AWSComputeSummary"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSComputeSummary"),
        migrations.RunSQL(sql=drop_matview_sql("AWSComputeSummaryByAccount"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSComputeSummaryByAccount"),
        migrations.RunSQL(sql=drop_matview_sql("AWSComputeSummaryByRegion"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSComputeSummaryByRegion"),
        migrations.RunSQL(sql=drop_matview_sql("AWSComputeSummaryByService"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSComputeSummaryByService"),
        migrations.RunSQL(sql=drop_matview_sql("AWSCostSummary"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSCostSummary"),
        migrations.RunSQL(sql=drop_matview_sql("AWSCostSummaryByAccount"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSCostSummaryByAccount"),
        migrations.RunSQL(sql=drop_matview_sql("AWSCostSummaryByRegion"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSCostSummaryByRegion"),
        migrations.RunSQL(sql=drop_matview_sql("AWSCostSummaryByService"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSCostSummaryByService"),
        migrations.RunSQL(sql=drop_matview_sql("AWSDatabaseSummary"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSDatabaseSummary"),
        migrations.RunSQL(sql=drop_matview_sql("AWSNetworkSummary"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSNetworkSummary"),
        migrations.RunSQL(sql=drop_matview_sql("AWSStorageSummary"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSStorageSummary"),
        migrations.RunSQL(sql=drop_matview_sql("AWSStorageSummaryByAccount"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSStorageSummaryByAccount"),
        migrations.RunSQL(sql=drop_matview_sql("AWSStorageSummaryByRegion"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSStorageSummaryByRegion"),
        migrations.RunSQL(sql=drop_matview_sql("AWSStorageSummaryByService"), reverse_sql=migrations.RunSQL.noop),
        migrations.DeleteModel(name="AWSStorageSummaryByService"),
```

**Review by Red-HAP — CHANGES_REQUESTED** [review]
Asking for a warm fuzzy: These views are not being hit from the background refresh process, yes?

**Comment by maskarb** [comment]
> Also asking for a warm fuzzy: These views are not being hit from the background refresh process, yes?

As far as I know, we don't touch these tables anymore. Unless there is some DB trigger that I am not aware of, it should be save to remove all of these views.

File: koku/reporting/migrations/0213_delete_aws_mat_views.py (hunk 1 above)
Comment (maskarb) [reply koku/reporting/migrations/0213_delete_aws_mat_views.py:4]: The `get_model` was not working for matviews, so I pass in the model name instead.

**Pull request closed without merging**